api/python/provisioner/commands/destroy_vm.py

```diff
@@ -0,0 +1,157 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+deploy_states = dict(
+    prvsnr=[
+        "system.storage.glusterfs.teardown.volume_remove",
+        "system.storage.glusterfs.teardown.remove_dir",
+        "system.storage.glusterfs.teardown.services",
+        "system.storage.glusterfs.teardown.package_remove"
+    ]
+)
+
+run_args_type = build_deploy_run_args(deploy_states)
+
+
+@attr.s(auto_attribs=True)
+class DestroyVM(Deploy):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = run_args_type
+
+    def _remove_prvsnr(self, nodes):
+        for node in nodes:
+            if local_minion_id() == node:
+                logger.info(f"remove cortx-prvsnr rpms on {node}")
+                run_subprocess_cmd(
+                    f"ssh {node} yum erase -y cortx-prvsnr")
+            else:
+                logger.info(
+                    f"remove cortx-prvsnr python36-cortx-prvsnr on {node}")
+                run_subprocess_cmd(
+                    f"ssh {node} yum erase -y"
+                    "cortx-prvsnr python36-cortx-prvsnr")
+
+    def _remove_passwordless_ssh(self, nodes):
+        for node in nodes:
+            logger.info(f"Remove passwordless configuration with {node}")
+            run_subprocess_cmd(
+                f"rm -rf /root/.ssh")
+
+    def _remove_salt(self, nodes):
+        salt_cache_dir = '/var/cache/salt'
+        salt_config = '/etc/salt'
+
+        for node in nodes:
+            logger.info(f"stopping salt-minion/salt-master on {node}")
+            run_subprocess_cmd(
+                f"ssh {node} systemctl stop salt-minion salt-master")
+
+            logger.info(f"remove salt rpms on {node}")
+            run_subprocess_cmd(
+                f"ssh {node} yum erase -y salt-minion salt-master salt")
+
+            logger.info(f"Removing salt from node {node}")
+            run_subprocess_cmd(f"ssh {node} rm -rf {salt_cache_dir}")
+            run_subprocess_cmd(f"ssh {node} rm -rf {salt_config}")
+
+    def _remove_dir(self, list_dir, nodes):
+        for node in nodes:
+            for di in list_dir:
+                logger.info(f"cleaning up {di} dir on {node}")
+                run_subprocess_cmd(f"ssh {node} rm -rf {di}")
+
+    def _run_states(self, states_group: str, run_args: run_args_type):
+        setup_type = run_args.setup_type
+        targets = run_args.targets
+        states = deploy_states[states_group]
+        stages = run_args.stages
+        primary = self._primary_id()
+        secondaries = f"not {primary}"
+        nodes = []
+
+        if states_group == 'prvsnr' and targets == ALL_MINIONS:
+            node_list = PillarKey('cluster/node_list')
+            pillar = PillarResolver(config.LOCAL_MINION).get([node_list])
+            pillar = next(iter(pillar.values()))
+            nodes = pillar[node_list]
+        else:
+            nodes.append(targets)
+
+        # apply states
+        for state in states:
+            if setup_type == SetupType.SINGLE:
```

Comment:
in case of provisioner gluster would be cleaned up here and salt - later, but it's a wrong order. teardown should go in opposite way comparing to deployment. please consider to implement states for all and list them in deploy_states

Reply:
Also, I have a doubt. Should we allow gluster on single node?
If not, then we just have to operate on salt packages.

Reply:
Added logic to ignore glusterfs in case of single node